models/org.go

```diff
@@ -250,13 +311,13 @@ func GetOrgByName(name string) (*User, error) {
 func CountOrganizations() int64 {
 	count, _ := db.GetEngine(db.DefaultContext).
 		Where("type=1").
-		Count(new(User))
+		Count(new(Organization))
 	return count
 }
 
 // DeleteOrganization completely and permanently deletes everything of organization.
-func DeleteOrganization(org *User) (err error) {
-	if !org.IsOrganization() {
+func DeleteOrganization(org *Organization) (err error) {
+	if org.Type != UserTypeOrganization {
 		return fmt.Errorf("%s is a user not an organization", org.Name)
 	}
```

Comment:
Give that you explicitly have to give a Organization, is this check still plausible?